Event/Event.gd

```diff
@@ -0,0 +1,11 @@
+extends Resource
+class_name Event
+
+var EVENTS_CLASSIFICATION = [Event_Mob, Event_Random, Event_Shop, Event_Heal]
```

Comment:
maybe a `const` would make more sense ? since it probably won't be modified at runtime.
Also i think it's lacking typing:
`var EVENTS_CLASSIFICATION: Array[Event] = [Event_Mob, Event_Random, Event_Shop, Event_Heal]`

Reply:
I'll need some help here, I get this compiler error yelling at me when I try to change it to a const

```
Line 4:Cannot have an element of type "Event_Mob" in an array of type "Array[EventBase]".
Line 4:Assigned value for constant "EVENTS_CLASSIFICATION" isn't a constant expression.
```

Reply:
ah probably because you can't have a const with an event that has values that can change

Reply:
Ok, so Godot's yelling at me at this new change:

```
var EVENTS_CLASSIFICATION: Array[EventBase] = [EventMob, EventRandom, EventShop, EventHeal]

Line 4:Cannot have an element of type "EventMob" in an array of type "Array[EventBase]".
```

Which is kind of weird, since all of those classes extend from EventBase. However, 
```
var EVENTS_CLASSIFICATION: Array[Resource] = [EventMob, EventRandom, EventShop, EventHeal]
```

Doesn't throw a compiler error. I'll leave it as that for now, unless you have a better recommendation.

Reply:
Yeah it doesn't know they are child of this class if not instantiated, I had the some problem the other day